lib/accesslists/hierarchy.go

```diff
@@ -543,7 +543,17 @@ func (s *Hierarchy) GetHierarchyForUser(ctx context.Context, accessList *accessl
 	if err != nil {
 		return nil, nil, trace.Wrap(err)
 	}
-	owners, err := s.expandOwnerOf(ctx, ancestors, user)
+
+	// We need to consider 2 scenarios:
+	// - An access list can be configured as 1 level owner of another access list
+	//   like access list A.Owners = [B], {B.members = [alice], B.Status.OwnerOf = [A]
+	//   In that case we need to explore the OwnerOf edges directly from input accessList and check if the user meets
+	//   the ownership requirements of the parent access list.
+	//
+	// - An access list can be configured as 2 level owner of another access list
```

Comment:
It can be more than 2 as well, right? Technically, it can be any depth (up to our 10 limit) I think. Is this scenario also handled now?

Reply:
Yes It is handled and I have added test coverage for this case: 

https://github.com/gravitational/teleport/pull/60036/files#diff-76c80292a091da3736e9bc34bc7390fce6b3389d85344541e884bb341af10adbR259-R264

To not limit to only 2 levels